pkg/proto/nbcontract/v1/utils.go

```diff
@@ -0,0 +1,289 @@
+	clusterVersion, _ := semver.Make(cs.Properties.OrchestratorProfile.OrchestratorVersion)
+	minVersion, _ := semver.Make("1.18.0")
+
+	var connectivityCheckCommand string
+	if clusterVersion.GTE(minVersion) {
+		connectivityCheckCommand = `curl -v --insecure --proxy-insecure https://` + registry + `/v2/`
+	} else {
+		connectivityCheckCommand = `nc -vz ` + registry + ` 443`
+	}
+
+	return connectivityCheckCommand
+}
+
+// GetOrderedKubeletConfigFlagString returns an ordered string of key/val pairs.
+// copied from AKS-Engine and filter out flags that already translated to config file.
+func GetKubeletConfigFlag(k map[string]string, cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile,
+	kubeletConfigFileToggleEnabled bool) map[string]string {
+	/* NOTE(mainred): kubeConfigFile now relies on CustomKubeletConfig, while custom configuration is not
+	compatible with CustomKubeletConfig. When custom configuration is set we want to override every
+	configuration with the customized one. */
+	kubeletCustomConfigurations := getKubeletCustomConfiguration(cs.Properties)
+	if kubeletCustomConfigurations != nil {
+		return getKubeletConfigFlagWithCustomConfiguration(kubeletCustomConfigurations, k)
+	}
+
+	if k == nil {
+		return nil
+	}
+	// Always force remove of dynamic-config-dir.
+	kubeletConfigFileEnabled := agent.IsKubeletConfigFileEnabled(cs, profile, kubeletConfigFileToggleEnabled)
+	kubeletConfigFlags := map[string]string{}
+	ommitedKubletConfigFlags := datamodel.GetCommandLineOmittedKubeletConfigFlags()
+	for key := range k {
+		if !kubeletConfigFileEnabled || !agent.TranslatedKubeletConfigFlags[key] {
+			if !ommitedKubletConfigFlags[key] {
+				kubeletConfigFlags[key] = k[key]
+			}
+		}
+	}
+	return kubeletConfigFlags
+}
+
+func getKubeletConfigFlagWithCustomConfiguration(customConfig, defaultConfig map[string]string) map[string]string {
+	config := customConfig
+
+	for k, v := range defaultConfig {
+		// add key-value only when the flag does not exist in custom config.
+		if _, ok := config[k]; !ok {
+			config[k] = v
+		}
+	}
+
+	ommitedKubletConfigFlags := datamodel.GetCommandLineOmittedKubeletConfigFlags()
+	for key := range config {
+		if ommitedKubletConfigFlags[key] {
+			delete(config, key)
+		}
+	}
+	return config
+}
+
+func getKubeletCustomConfiguration(properties *datamodel.Properties) map[string]string {
+	if properties.CustomConfiguration == nil || properties.CustomConfiguration.KubernetesConfigurations == nil {
+		return nil
+	}
+	kubeletConfigurations, ok := properties.CustomConfiguration.KubernetesConfigurations["kubelet"]
+	if !ok {
+		return nil
+	}
+	if kubeletConfigurations.Config == nil {
+		return nil
+	}
+	// empty config is treated as nil.
+	if len(kubeletConfigurations.Config) == 0 {
+		return nil
+	}
+	return kubeletConfigurations.Config
+}
+
+func ValidateAndSetLinuxKubeletFlags(kubeletFlags map[string]string, cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile) {
```

Comment:
Just to confirm, this function is not really used in the current PR, right? Is it supposed to be used by AKS-RP later?

Reply:
see above.